pkg/controllers/nodeclaim/termination/controller.go

```diff
@@ -89,7 +89,11 @@ func (c *Controller) Finalize(ctx context.Context, nodeClaim *v1beta1.NodeClaim)
 		return reconcile.Result{}, nil
 	}
 	if nodeClaim.Status.ProviderID != "" {
-		if err = c.cloudProvider.Delete(ctx, nodeClaim); cloudprovider.IgnoreNodeClaimNotFoundError(err) != nil || cloudprovider.IsRetryableError(err) {
+		err = c.cloudProvider.Delete(ctx, nodeClaim)
```

Comment:
style: consider the following

```go
if err := c.cloudProvider.Delete(ctx, nodeClaim); cloudprovider.IgnoreNodeClaimNotFound(err) != nil {
  if cloudprovider.IsRetryableError(err) {
    return reconcile.Result{RequeueAfter: 10*time.Second}, nil
  }
  return reconcile.Result{}, fmt.Errorf("terminating cloudprovider instance, %v", err)
}
```

Reply:
Consider also adding a comment on why we are retrying here with this kind of error, since it isn't immediately apparent to the reader what a "retryable error" is